lib/checken/schema.rb

```diff
@@ -144,5 +107,63 @@ def schema
       @schema.sort.to_h
     end
 
+    private
+
+    def handle_strict_permission_check!(permission_path, user_proxy, object)
+      permissions = @root_group.find_permissions_from_path(permission_path)
+
+      if permissions.size == 1
+        # If we only have a single permission, we'll just run the check
+        # as normal through the check process. This will work as normal and raise
+        # and return directly.
+        permissions.first.check!(user_proxy, object)
+
+      elsif permissions.size == 0
+        # No permissions found
+        raise Checken::NoPermissionsFoundError, "No permissions found matching #{permission_path}"
+
+      else
+        # If we have multiple permissions, we need to loop through each permission
+        # and handle them as appropriate.
+        granted_permissions = []
+        ungranted_permissions = 0
+        permissions.each do |permission|
+          begin
+            permission.check!(user_proxy, object).each do |permission|
+              granted_permissions << permission
+            end
+          rescue Checken::PermissionDeniedError => e
+            if e.code == 'PermissionNotGranted'
+              # If the permission isn't granted, update the counter so we can
+              # keep track of the number of ungranted permissions.
+              ungranted_permissions += 1
+            else
+              # Raise other errors as normal
+              raise
+            end
+          end
+        end
+
+        if permissions.size == ungranted_permissions
+          # If the user is ungranted to all the found permissions, they do not
+          # have access and should be denied.
+          raise PermissionDeniedError.new('PermissionNotGranted', "User does not have any permissions #{permissions.map(&:path).join(', ')} permission.", permissions.first)
+        else
+          granted_permissions
+        end
+      end
+    end
+
+    def handle_unstrict_permission_check!(permission_path, user_proxy)
+      unless user_proxy.is_a?(Checken::UserProxy)
+        user_proxy = @group.schema.config.user_proxy_class.new(user_proxy)
```

Comment:
Should this not be `@root_group`?

Reply:
Nice catch. 

So this can actually be simplified to just `config.user_proxy_class.new(user_proxy)` since this method lives on the Schema.